src/sagemaker/model.py

```diff
@@ -187,7 +188,9 @@ def prepare_container_def(self, instance_type):  # pylint disable=unused-argumen
     def _upload_code(self, key_prefix):
         local_code = get_config_value('local.local_code', self.sagemaker_session.config)
         if self.sagemaker_session.local_mode and local_code:
-            self.uploaded_code = None
+            script_name = self.entry_point
+            dir_name = 'file://' + self.source_dir
+            self.uploaded_code = UploadedCode(s3_prefix=dir_name, script_name=script_name)
```

Comment:
I think I am confused. It looks like we have potentially a local uri being set inside of a field called s3_prefix.

Reply:
Talked offline. We need @iquintero to have a look.

Reply:
Reverted this change after talking with @iquintero 

I could create a fake uploaded_code in the fake model first. This won't trigger local mode at all.